src/compiler/checker.ts

```diff
@@ -35579,14 +35603,17 @@ namespace ts {
                 reportImplicitAny(node, anyType);
             }
 
-            const type = getTypeFromMappedTypeNode(node) as MappedType;
-            const nameType = getNameTypeFromMappedType(type);
-            if (nameType) {
-                checkTypeAssignableTo(nameType, keyofConstraintType, node.nameType);
-            }
-            else {
-                const constraintType = getConstraintTypeFromMappedType(type);
-                checkTypeAssignableTo(constraintType, keyofConstraintType, getEffectiveConstraintOfTypeParameter(node.typeParameter));
+            const type = getTypeFromMappedTypeNode(node);
+            // Continue to check if the type returned is a mapped type, that means it wasn't resolved to a homomorphic tuple type

```

Comment:
where does this typechecking for a homomorphic mapped type happen now? in the added `instantiateMappedTupleType` call?

Reply:
So in case the returned type is a tuple type (which I now realize it may be a simpler condition here) we don't have a mapped type to check, one condition is implicitly checked, as the type can only be homomorphic if it has no nameType (the `AS rest` component of the type parameter) the other on assignability of the constraint to the keyOfConstraintType should be once again implicit given the constraint is the keyof of a tuple BUT I may be wrong :)